rita_common/src/payment_validator/mod.rs

```diff
@@ -115,24 +164,15 @@ pub fn calculate_unverified_payments(router: Identity) -> Uint256 {
 /// This endpoint specifically (and only this one) is fully idempotent so that we can retry
 /// txid transmissions
 pub fn validate_later(ts: ToValidate) {
-    let mut history = HISTORY.write().unwrap();
-    if let Some(txid) = ts.payment.txid.clone() {
-        if !history.successful_transactions.contains(&txid) {
-            // insert is safe to run multiple times just so long as we check successful tx's for duplicates
-            history.unvalidated_transactions.insert(ts);
-        }
-    } else {
-        error!(
-            "Someone tried to insert an unpublished transaction to validate!? {:?}",
-            ts
-        );
-        // in a development env we want to draw attention to this case
-        #[cfg(feature = "development")]
```

Comment:
are you sure this is checked later? I'm not sure.

Reply:
Are you saying txid is none?, Im checking that here in the same function
Or are you saying printing this log in development? If so ill add that

Reply:
Are you saying txid is none? Looks likes im currently just printing a log, Ill add a fix here where we dont add the tx to the unvalidated set